### PR TITLE
Document i18n locale limit

### PR DESCRIPTION
**Pull request by ijjk**
Add check for i18n locale count and ensure the limit is documented

## Documentation / Examples

- [x] Make sure the linting passes

### CODE REVIEW

**Review by styfle — DISMISSED** [review]

File: packages/next/server/config.ts [hunk 1]
```diff
@@ -320,6 +320,12 @@ function assignDefaults(userConfig: { [key: string]: any }) {
       )
     }
 
+    if (i18n.locales.length > 100) {
+      throw new Error(
+        `Received ${i18n.locales.length} i18n.locales items which exceeds the max of 100, please reduce the number of items to continue.\nSee more info here: https://nextjs.org/docs/messages/invalid-i18n-config`
```
Comment (styfle) [inline packages/next/server/config.ts:325]: This error page doesn't mention the limit

Comment (ijjk) [reply packages/next/server/config.ts:325]: Good point, added it in that doc as well

**Review by styfle — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| buildDuration | 13s | 12.9s | -85ms |
| buildDurationCached | 3s | 2.8s | -154ms |
| nodeModulesSize | 50.4 MB | 50.4 MB | ⚠️ +733 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.114 | 2.217 | ⚠️ +0.1 |
| / avg req/sec | 1182.55 | 1127.82 | ⚠️ -54.73 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.159 | 1.232 | ⚠️ +0.07 |
| /error-in-render avg req/sec | 2156.87 | 2029.54 | ⚠️ -127.33 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| 359.HASH.js gzip | 2.96 kB | 2.96 kB | ✓ |
| 745.HASH.js gzip | 180 B | 180 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 21 kB | 21 kB | ✓ |
| webpack-HASH.js gzip | 1.53 kB | 1.53 kB | ✓ |
| Overall change | 67.9 kB | 67.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| _app-HASH.js gzip | 803 B | 803 B | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 554 B | 554 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.52 kB | 2.52 kB | ✓ |
| head-HASH.js gzip | 2.28 kB | 2.28 kB | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| image-HASH.js gzip | 5.63 kB | 5.63 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 319 B | 319 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 19.1 kB | 19.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| _buildManifest.js gzip | 489 B | 489 B | ✓ |
| Overall change | 489 B | 489 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 544 B | 544 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| buildDuration | 9.9s | 10.2s | ⚠️ +321ms |
| buildDurationCached | 4.1s | 4s | -91ms |
| nodeModulesSize | 50.4 MB | 50.4 MB | ⚠️ +733 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.148 | 2.175 | ⚠️ +0.03 |
| / avg req/sec | 1164.06 | 1149.37 | ⚠️ -14.69 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.235 | 1.242 | ⚠️ +0.01 |
| /error-in-render avg req/sec | 2024.93 | 2012.27 | ⚠️ -12.66 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| 17.HASH.js gzip | 2.98 kB | 2.98 kB | ✓ |
| 18.HASH.js gzip | 185 B | 185 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 8.4 kB | 8.4 kB | ✓ |
| webpack-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| Overall change | 68.5 kB | 68.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| _app-HASH.js gzip | 791 B | 791 B | ✓ |
| _error-HASH.js gzip | 3.76 kB | 3.76 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 3.02 kB | 3.02 kB | ✓ |
| withRouter-HASH.js gzip | 294 B | 294 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 17.6 kB | 17.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| _buildManifest.js gzip | 500 B | 500 B | ✓ |
| Overall change | 500 B | 500 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| index.html gzip | 577 B | 577 B | ✓ |
| link.html gzip | 588 B | 588 B | ✓ |
| withRouter.html gzip | 569 B | 569 B | ✓ |
| Overall change | 1.73 kB | 1.73 kB | ✓ |

</details>

</details>
Commit: 2be843b04fdfcf5b0d9efc525502201ffff7de32

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| buildDuration | 13.5s | 14.9s | ⚠️ +1.4s |
| buildDurationCached | 3.1s | 3s | -49ms |
| nodeModulesSize | 50.4 MB | 50.4 MB | ⚠️ +733 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.491 | 2.227 | -0.26 |
| / avg req/sec | 1003.67 | 1122.8 | +119.13 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.338 | 1.259 | -0.08 |
| /error-in-render avg req/sec | 1868.29 | 1984.94 | +116.65 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| 359.HASH.js gzip | 2.96 kB | 2.96 kB | ✓ |
| 745.HASH.js gzip | 180 B | 180 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 21 kB | 21 kB | ✓ |
| webpack-HASH.js gzip | 1.53 kB | 1.53 kB | ✓ |
| Overall change | 67.9 kB | 67.9 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.1 kB | 31.1 kB | ✓ |
| Overall change | 31.1 kB | 31.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| _app-HASH.js gzip | 803 B | 803 B | ✓ |
| _error-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| amp-HASH.js gzip | 554 B | 554 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.52 kB | 2.52 kB | ✓ |
| head-HASH.js gzip | 2.28 kB | 2.28 kB | ✓ |
| hooks-HASH.js gzip | 903 B | 903 B | ✓ |
| image-HASH.js gzip | 5.63 kB | 5.63 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 319 B | 319 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 19.1 kB | 19.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| _buildManifest.js gzip | 489 B | 489 B | ✓ |
| Overall change | 489 B | 489 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| index.html gzip | 531 B | 531 B | ✓ |
| link.html gzip | 544 B | 544 B | ✓ |
| withRouter.html gzip | 525 B | 525 B | ✓ |
| Overall change | 1.6 kB | 1.6 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| buildDuration | 10.3s | 10.7s | ⚠️ +451ms |
| buildDurationCached | 4.5s | 4s | -443ms |
| nodeModulesSize | 50.4 MB | 50.4 MB | ⚠️ +733 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.402 | 2.349 | -0.05 |
| / avg req/sec | 1040.78 | 1064.44 | +23.66 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.326 | 1.357 | ⚠️ +0.03 |
| /error-in-render avg req/sec | 1885 | 1842.81 | ⚠️ -42.19 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| 17.HASH.js gzip | 2.98 kB | 2.98 kB | ✓ |
| 18.HASH.js gzip | 185 B | 185 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 8.4 kB | 8.4 kB | ✓ |
| webpack-HASH.js gzip | 1.22 kB | 1.22 kB | ✓ |
| Overall change | 68.5 kB | 68.5 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31.3 kB | 31.3 kB | ✓ |
| Overall change | 31.3 kB | 31.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| _app-HASH.js gzip | 791 B | 791 B | ✓ |
| _error-HASH.js gzip | 3.76 kB | 3.76 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.71 kB | 2.71 kB | ✓ |
| head-HASH.js gzip | 2.97 kB | 2.97 kB | ✓ |
| hooks-HASH.js gzip | 911 B | 911 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 3.02 kB | 3.02 kB | ✓ |
| withRouter-HASH.js gzip | 294 B | 294 B | ✓ |
| e025d2764813..52f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 17.6 kB | 17.6 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| _buildManifest.js gzip | 500 B | 500 B | ✓ |
| Overall change | 500 B | 500 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | ijjk/next.js docs/i18n-limit | Change |
| - | - | - | - |
| index.html gzip | 577 B | 577 B | ✓ |
| link.html gzip | 588 B | 588 B | ✓ |
| withRouter.html gzip | 569 B | 569 B | ✓ |
| Overall change | 1.73 kB | 1.73 kB | ✓ |

</details>

</details>
Commit: d108bb9ab333d408300ca55362151866edfc8bdb

**Pull request merged**